### DISCUSSION TITLE
[Blade Templates] How to show associated blog content from a tag?

### DISCUSSION THREAD

**Discussion by MeikyuuTrader** [statamic/cms#4775] [Q&A] [answered]
Using this [Statamic Blade](https://github.com/edalzell/statamic-blade), I'm trying to show blog entries that're linked to a tag. From the [Statamic youtube tutorial](https://www.youtube.com/watch?v=wQTqcYRLQbM), it uses `{{ entries }}` to get all the associated blog content in blog/tags/show.antlers.html. Url from blog/tags/tag1 in the show.blade.html from that path 

However, I'm using blade. Using `{{ dd(dump(get_defined_vars())) }}` I can see there's a `entries` variable, where the entries_count is the same as how many linked values there are. 

From looking at the dump, `entries` is a `Statamic\Stache\Query\EntryQueryBuilder` type. Is there anyway to get the inner data, ie the content, using Laravel Blade?

Although I could do a full on query, for each tag, get all the blog entries from the collection and only show ones with the current tag.

**Comment by jasonvarga** [comment]
Use the `@collection` directive provided by that addon.

```blade
@collection('pages', ['title:is' => 'My Title', 'author:is' => 'Erin', 'limit' => 3, 'sort' => 'title:desc'])\
    @if($entry['no_results'])
        <p>There are no results</p>
    @else
        {{ $entry['title'] }}
    @endif
@endcollection
```

**Comment by MeikyuuTrader** [comment, marked as answer]
Hmm, I wanted to avoid doing a collection query when I'm already in the blog/tags/show.blade.php page and use `entries` like the statamic video did. Since entries does exist there and it shows the correct count for all associated values with that said tag. 

```
    @collection('blog', ['taxonomy:tags' => "$slug", 'sort' => 'date:desc'])
        {{ dd($entry) }}
    @endcollection
```

^^ Just adding the correct query for Statamic Blade Plugin here incase someone else needs to do the same. 

# Edit Real Answer
I figured out a better solution. Since it's an EntryQueryBuilder, we can do all the same commmands as [Content Queries](https://statamic.dev/content-queries). Which means we can run `entries->get()` to get all the associated blog articles. ie

```
    @php
        $data = $entries->get();
    @endphp

    @foreach ($data as $ele)
        @data($ele)
            {{ $title }}
            {{ $url }}
        @enddata
    @endforeach
```

Thus there's no need to requery for ALL the blog entries, this would just get the blog entries linked to that tag.
